### DISCUSSION TITLE
NLDAS expansion

### DISCUSSION THREAD

**Discussion by hcorson-dosch-usgs** [DOI-USGS/lake-temperature-process-models#57] [General]
Just ran the NLDAS models for the full footprint (9,369 lakes). It ran in just under an hour. ~I'm generating the output files now.~ UPDATE: all 9333 NLDAS feather files are in `3_extract/out`. They are also zipped into a single zip file.

Out of 9369 runs, 9333 succeeded. 36 failed. 99.6% success rate. The 11 runs that had failed in [the MN run](https://github.com/USGS-R/lake-temperature-process-models/pull/48#issuecomment-1128137907) also failed this time around.

The mean lake depth across the 36 lakes for which NLDAS runs failed is 1.6 meters. Median lake depth is 0.96 m, minimum lake depth is 0 m (`nhdhr_151870592`), maximum lake depth is 14 m (`nhdhr_157360156`).

Details on the failed runs:
```r
> nrow(filter(p2_nldas_glm_uncalibrated_runs, glm_success==FALSE))
[1] 36
> p2_nldas_glm_uncalibrated_runs %>% filter(glm_success==FALSE) %>% select(site_id, driver, extraction_error, param_sim_stop, max_output_date, glm_code) %>% slice(1:20)
# A tibble: 20 × 6
   site_id       driver extraction_error param_sim_stop max_output_date glm_code
   <chr>         <chr>  <chr>            <chr>          <chr>              <int>
 1 nhdhr_109946… NLDAS  NA values        2022-04-11     2022-04-11             0
 2 nhdhr_114545… NLDAS  NA values        2022-04-11     2022-04-11             0
 3 nhdhr_117648… NLDAS  NA values        2022-04-11     2022-04-11             0
 4 nhdhr_120019… NLDAS  NA values        2022-04-11     2022-04-11             0
 5 nhdhr_120020… NLDAS  NA               2022-04-11     1989-02-23             1
 6 nhdhr_122546… NLDAS  NA values        2022-04-11     2022-04-11             0
 7 nhdhr_123165… NLDAS  NA values        2022-04-11     2022-04-11             0
 8 nhdhr_124617… NLDAS  NA values        2022-04-11     2022-04-11             0
 9 nhdhr_133099… NLDAS  NA values        2022-04-11     2022-04-11             0
10 nhdhr_144272… NLDAS  NA values        2022-04-11     2022-04-11             0
11 nhdhr_148621… NLDAS  NA values        2022-04-11     2022-04-11             0
12 nhdhr_151870… NLDAS  NA values        2022-04-11     2022-04-11             0
13 nhdhr_151949… NLDAS  NA values        2022-04-11     2022-04-11             0
14 nhdhr_151959… NLDAS  NA values        2022-04-11     2022-04-11             0
15 nhdhr_152517… NLDAS  NA values        2022-04-11     2022-04-11             0
16 nhdhr_155352… NLDAS  NA values        2022-04-11     2022-04-11             0
17 nhdhr_155639… NLDAS  NA values        2022-04-11     2022-04-11             0
18 nhdhr_157360… NLDAS  NA               2022-04-11     2022-04-11           134
19 nhdhr_159101… NLDAS  NA values        2022-04-11     2022-04-11             0
20 nhdhr_258611… NLDAS  NA values        2022-04-11     2022-04-11             0
> p2_nldas_glm_uncalibrated_runs %>% filter(glm_success==FALSE) %>% select(site_id, driver, extraction_error, param_sim_stop, max_output_date, glm_code) %>% slice(21:36)
# A tibble: 16 × 6
   site_id       driver extraction_error param_sim_stop max_output_date glm_code
   <chr>         <chr>  <chr>            <chr>          <chr>              <int>
 1 nhdhr_259216… NLDAS  NA values        2022-04-11     2022-04-11             0
 2 nhdhr_261524… NLDAS  NA values        2022-04-11     2022-04-11             0
 3 nhdhr_261577… NLDAS  NA values        2022-04-11     2022-04-11             0
 4 nhdhr_474867… NLDAS  NA values        2022-04-11     2022-04-11             0
 5 nhdhr_487840… NLDAS  NA values        2022-04-11     2022-04-11             0
 6 nhdhr_564130… NLDAS  NA values        2022-04-11     2022-04-11             0
 7 nhdhr_597459… NLDAS  NA values        2022-04-11     2022-04-11             0
 8 nhdhr_604470… NLDAS  NA values        2022-04-11     2022-04-11             0
 9 nhdhr_604559… NLDAS  NA values        2022-04-11     2022-04-11             0
10 nhdhr_703344… NLDAS  NA values        2022-04-11     2022-04-11             0
11 nhdhr_756519… NLDAS  NA values        2022-04-11     2022-04-11             0
12 nhdhr_809976… NLDAS  NA values        2022-04-11     2022-04-11             0
13 nhdhr_810037… NLDAS  NA values        2022-04-11     2022-04-11             0
14 nhdhr_810037… NLDAS  NA values        2022-04-11     2022-04-11             0
15 nhdhr_829559… NLDAS  NA values        2022-04-11     2022-04-11             0
16 nhdhr_916842… NLDAS  NA values        2022-04-11     2022-04-11             0
>  dput(p2_nldas_glm_uncalibrated_runs %>% filter(glm_success==FALSE) %>% pull(site_id))
c("nhdhr_109946618", "nhdhr_114545137", "nhdhr_117648183", "nhdhr_120019262",
"nhdhr_120020716", "nhdhr_122546701", "nhdhr_123165634", "nhdhr_124617242",
"nhdhr_133099150", "nhdhr_144272629", "nhdhr_148621069", "nhdhr_151870592",
"nhdhr_151949010", "nhdhr_151959489", "nhdhr_152517749", "nhdhr_155352079",
"nhdhr_155639566", "nhdhr_157360156", "nhdhr_159101759", "nhdhr_25861152",
"nhdhr_25921640", "nhdhr_26152472", "nhdhr_26157758", "nhdhr_47486745",
"nhdhr_48784080", "nhdhr_56413007", "nhdhr_59745939", "nhdhr_60447081",
"nhdhr_60455965", "nhdhr_70334497", "nhdhr_75651917", "nhdhr_80997655",
"nhdhr_81003721", "nhdhr_81003725", "nhdhr_82955915", "nhdhr_91684249"
)

```

**Comment by hcorson-dosch-usgs** [comment]
With updated NLDAS drivers (see [this issue](https://github.com/USGS-R/lake-temperature-model-prep/issues/341)), runs for 37 sites failed. There is one new failed site: `nhdhr_75654579`
```R
> nrow(filter(p2_nldas_glm_uncalibrated_runs, glm_success==FALSE))
[1] 37
> p2_nldas_glm_uncalibrated_runs %>% filter(glm_success==FALSE) %>% select(site_id, driver, extraction_error, param_sim_stop, max_output_date, glm_code) %>% slice(1:20)
# A tibble: 20 × 6
   site_id       driver extraction_error param_sim_stop max_output_date glm_code
   <chr>         <chr>  <chr>            <chr>          <chr>              <int>
 1 nhdhr_109946… NLDAS  NA values        2022-04-11     2022-04-11             0
 2 nhdhr_114545… NLDAS  NA values        2022-04-11     2022-04-11             0
 3 nhdhr_117648… NLDAS  NA values        2022-04-11     2022-04-11             0
 4 nhdhr_120019… NLDAS  NA values        2022-04-11     2022-04-11             0
 5 nhdhr_120020… NLDAS  NA               2022-04-11     1989-02-23             1
 6 nhdhr_122546… NLDAS  NA values        2022-04-11     2022-04-11             0
 7 nhdhr_123165… NLDAS  NA values        2022-04-11     2022-04-11             0
 8 nhdhr_124617… NLDAS  NA values        2022-04-11     2022-04-11             0
 9 nhdhr_133099… NLDAS  NA values        2022-04-11     2022-04-11             0
10 nhdhr_144272… NLDAS  NA values        2022-04-11     2022-04-11             0
11 nhdhr_148621… NLDAS  NA values        2022-04-11     2022-04-11             0
12 nhdhr_151870… NLDAS  NA values        2022-04-11     2022-04-11             0
13 nhdhr_151949… NLDAS  NA values        2022-04-11     2022-04-11             0
14 nhdhr_151959… NLDAS  NA values        2022-04-11     2022-04-11             0
15 nhdhr_152517… NLDAS  NA values        2022-04-11     2022-04-11             0
16 nhdhr_155352… NLDAS  NA values        2022-04-11     2022-04-11             0
17 nhdhr_155639… NLDAS  NA values        2022-04-11     2022-04-11             0
18 nhdhr_157360… NLDAS  NA               2022-04-11     2022-04-11           134
19 nhdhr_159101… NLDAS  NA values        2022-04-11     2022-04-11             0
20 nhdhr_258611… NLDAS  NA values        2022-04-11     2022-04-11             0
> p2_nldas_glm_uncalibrated_runs %>% filter(glm_success==FALSE) %>% select(site_id, driver, extraction_error, param_sim_stop, max_output_date, glm_code) %>% slice(21:37)
# A tibble: 17 × 6
   site_id       driver extraction_error param_sim_stop max_output_date glm_code
   <chr>         <chr>  <chr>            <chr>          <chr>              <int>
 1 nhdhr_259216… NLDAS  NA values        2022-04-11     2022-04-11             0
 2 nhdhr_261524… NLDAS  NA values        2022-04-11     2022-04-11             0
 3 nhdhr_261577… NLDAS  NA values        2022-04-11     2022-04-11             0
 4 nhdhr_474867… NLDAS  NA values        2022-04-11     2022-04-11             0
 5 nhdhr_487840… NLDAS  NA values        2022-04-11     2022-04-11             0
 6 nhdhr_564130… NLDAS  NA values        2022-04-11     2022-04-11             0
 7 nhdhr_597459… NLDAS  NA values        2022-04-11     2022-04-11             0
 8 nhdhr_604470… NLDAS  NA values        2022-04-11     2022-04-11             0
 9 nhdhr_604559… NLDAS  NA values        2022-04-11     2022-04-11             0
10 nhdhr_703344… NLDAS  NA values        2022-04-11     2022-04-11             0
11 nhdhr_756519… NLDAS  NA values        2022-04-11     2022-04-11             0
12 nhdhr_756545… NLDAS  NA values        2022-04-11     2022-04-11             0
13 nhdhr_809976… NLDAS  NA values        2022-04-11     2022-04-11             0
14 nhdhr_810037… NLDAS  NA values        2022-04-11     2022-04-11             0
15 nhdhr_810037… NLDAS  NA values        2022-04-11     2022-04-11             0
16 nhdhr_829559… NLDAS  NA values        2022-04-11     2022-04-11             0
17 nhdhr_916842… NLDAS  NA values        2022-04-11     2022-04-11             0

> dput(p2_nldas_glm_uncalibrated_runs %>% filter(glm_success==FALSE) %>% pull(site_id))
c("nhdhr_109946618", "nhdhr_114545137", "nhdhr_117648183", "nhdhr_120019262",
"nhdhr_120020716", "nhdhr_122546701", "nhdhr_123165634", "nhdhr_124617242",
"nhdhr_133099150", "nhdhr_144272629", "nhdhr_148621069", "nhdhr_151870592",
"nhdhr_151949010", "nhdhr_151959489", "nhdhr_152517749", "nhdhr_155352079",
"nhdhr_155639566", "nhdhr_157360156", "nhdhr_159101759", "nhdhr_25861152",
"nhdhr_25921640", "nhdhr_26152472", "nhdhr_26157758", "nhdhr_47486745",
"nhdhr_48784080", "nhdhr_56413007", "nhdhr_59745939", "nhdhr_60447081",
"nhdhr_60455965", "nhdhr_70334497", "nhdhr_75651917", "nhdhr_75654579",
"nhdhr_80997655", "nhdhr_81003721", "nhdhr_81003725", "nhdhr_82955915",
"nhdhr_91684249")
```

**Reply by jordansread** [reply]
Oh no. That is unexpected and shocking. 😱

**Reply by jordansread** [reply]
Well...I did misread this originally and think that the count was now 37 failures when we originally had 11 failures (which I think was the MN count). But now I see this is for the whole footprint and we're instead going from 36 to 37. So not quite as bad, but still unexpected to me. I'd expect the failures to either stay the same or decrease with this input data modification. I'm in favor of keeping the change as-is regardless though, as having mixed precipitation in the inputs did cause odd issues in the past that weren't necessarily model failures caught by the codes, but were unrealistic.

**Comment by hcorson-dosch-usgs** [comment]
Surface evaluation plots for the 5/25/2022 run (9,332 lakes succeeded) are [here](https://doimspp.sharepoint.com/sites/IIDDStaff/Shared%20Documents/Project%20-%20Modeling%20Lakes/GLM/lake-temperature-process-models/NLDAS_surface_evaluation_plots/FullFootprint_20220526). There were 3,665 evaluation sites (sites with >= 10 `min_obs_dates` and successful model runs)

**Comment by hcorson-dosch-usgs** [comment]
@lindsayplatt @jread-usgs 

Just ran the NLDAS models for the full footprint (with [landsat Kd](https://github.com/USGS-R/lake-temperature-model-prep/pull/343) and [updated WQP data](https://github.com/USGS-R/lake-temperature-model-prep/pull/344)). It took 130 minutes to run the models (`p2_nldas` targets). Out of 14,617 sites, models for 189 lakes failed. 2 failed with `glm_code` = 1 and `max_output_date` < `param_sim_stop` (no extraction error). 187 failed with extraction errors:
```r
> p2_nldas_glm_uncalibrated_runs %>% group_by(extraction_error) %>% summarize(n=n())
# A tibble: 4 × 2
  extraction_error                                                   n
  <chr>                                                              <int>
1 NA temperature values                                                119
2 Number of layers drops to 1                                           67
3 Number of layers drops to 1, Maximum hice value exceeds lake depth     1
4 NA                                                                 14430
```
Note that the 'NA temperature values' error is triggered by [the extraction of GLM temperatures](https://github.com/USGS-R/lake-temperature-process-models/blob/main/2_run/src/run_glm3.R#L17) and precedes [the generation of the other two extraction errors](https://github.com/USGS-R/lake-temperature-process-models/blob/main/2_run/src/run_glm3.R#L33-L48). So lakes for which temperature data extraction failed (`NA temperature values`) may have also have excessive ice generation or only 1 simulation layer. I could re-work that code so that the # of layers and ice thickness are checked _before_ temperature data is extracted.

I'm going to generate the surface prediction evaluation plots for these runs now.

**Reply by hcorson-dosch-usgs** [reply]
40/189 of the failed sites were sites included in the previous extent of the full footprint (# lakes = [9369](https://github.com/USGS-R/lake-temperature-model-prep/pull/340)). As noted in my post from 5/25, we were before noting 37 failed model runs. So an additional 3 failures were caught with the addition of the [# layers check and ice thickness check](https://github.com/USGS-R/lake-temperature-process-models/blob/main/2_run/src/run_glm3.R#L33-L48) _OR_ were genuinely new failures due to some change in the `nml_list.rds` data for those lakes.

**Reply by lindsayplatt** [reply]
This is a 98.7% success rate 🎉 Nice job scaling up! When we're modeling 14k+ lakes and ~1% fail, I'm not sure it is worth the effort to really dive deep here (esp given our time constraint & the fact that we kind of know why they failed thanks to your error handling. I will also let @jread-usgs comment on that. 

Maybe do a quick check for the lake depth of these 189 failed sites to see if they fall in that shallow category.

**Reply by jordansread** [reply]
It is a great sign for the modeling pipeline that nearly doubling the lakes modeled was accomplished relatively easily - nice work Hayley!

I agree that we've got a pretty good sense of why these would fail and a good set of catches to avoid exporting them. I'm good with moving on after Lindsay's suggestion of checking the distribution of depths for the 189.

**Reply by hcorson-dosch-usgs** [reply]
Ok great!

The mean lake depth across the 189 lakes for which NLDAS runs failed is 0.7 meters. Median lake depth is 0.6 m, minimum lake depth is 0 m (4 sites - `c("nhdhr_113236931", "nhdhr_151870592", "nhdhr_151874009", "nhdhr_45361479"
)`), maximum lake depth is 9.1 m (`nhdhr_76157357`).

^ Pretty consistent with the pattern we've seen so far of failures being concentrated in very shallow lakes.

The NLDAS surface evaluation plots are [here](https://doimspp.sharepoint.com/:f:/r/sites/IIDDStaff/Shared%20Documents/Project%20-%20Modeling%20Lakes/GLM/lake-temperature-process-models/NLDAS_surface_evaluation_plots/FullFootprint_20220601?csf=1&web=1&e=czmbG0)
. There were 3,753 evaluation sites (sites with >= 10 min_obs_dates and successful model runs) -- up from 3,665 previously (so not much of an increase - to be expected, I think, given the changes to the pipeline).

**Reply by hcorson-dosch-usgs** [reply]
NLDAS model evaluation plots for surface and bottom predictions are [HERE](https://doimspp.sharepoint.com/:f:/r/sites/IIDDStaff/Shared%20Documents/Project%20-%20Modeling%20Lakes/GLM/lake-temperature-process-models/NLDAS_evaluation_plots/FullFootprint_20220601?csf=1&web=1&e=dWdfQa). Now that we are only evaluating predictions for lakes within CASC states we're down to 3688 evaluation sites.

**Reply by hcorson-dosch-usgs** [reply]
Now that we are also restricting model runs and therefore extracted output to lakes within CASC states, our success percentage is 98.6% (NLDAS models for 12687/12864 lakes succeeded, models for 177 lakes failed).

**Comment by lindsayplatt** [comment]
Using the latest NLDAS-driven GLM build completed on June 16, 2022, the full footprint of thermal metrics has been completed (built with `lake-temperature-out` repo using code changes from [this PR](https://github.com/USGS-R/lake-temperature-out/pull/62)). The `https://www.sciencebase.gov/catalog/item/6206d3c2d34ec05caca53071` file [on our ScienceBase item](https://www.sciencebase.gov/catalog/item/6206d3c2d34ec05caca53071) was also added. In the thread, I will add a few graphics for summarizing the results.

**Reply by lindsayplatt** [reply]
The code in this thread assumes you have the `annual_metrics_glm3_pb0nldas.csv` file in your current working directory (download from SB using the link above if you need it).

![image](https://user-images.githubusercontent.com/13220910/182669263-b1213c54-c5ea-4783-bec0-ef4c7f2bebb4.png)

```r
library(tidyverse)
library(sf)
nldas_metrics_all <- readr::read_csv('annual_metrics_glm3_pb0nldas.csv')

select_metrics <- nldas_metrics_all %>% 
  select(year, 
         starts_with("max_surf_"),
         starts_with("max_bot_")) %>% 
  pivot_longer(cols = c(!year)) %>% 
  separate(name, c("stat", "depth", "month_abb"), "_") %>% 
  filter(month_abb != "JulAugSep") %>% 
  mutate(month_num = as.numeric(factor(month_abb, levels = tolower(month.abb))))%>% 
  mutate(depth_f = factor(depth, levels=c('surf', 'bot'), labels=c('surface', 'bottom'), ordered=TRUE))

# Add groups by decade (the 2020 decade will have less data than the rest)
select_metrics_yearsgrp <- select_metrics %>% 
  mutate(decade = year - year%%10)

ggplot(select_metrics_yearsgrp, aes(y = value, group=decade)) + 
  geom_boxplot(outlier.colour = "darkgrey", outlier.shape = 1, outlier.alpha = 0.1) +
  scale_x_discrete(breaks=1:12, labels = month.abb) +
  facet_grid(depth_f ~ month_num) +
  ylab('GLM-predicted temperature, deg C') +
  ggtitle('Annual maximum temps at surface and bottom depths for each month of the year',
          subtitle = '     There is a boxplot per decade')
```

**Reply by lindsayplatt** [reply]
Most of the 247 lakes with a really high winter temperature are toward the southern part of our footprint. There is one lake near the WI-IL border `nhdhr_121624561` that may be suspicious.

![image](https://user-images.githubusercontent.com/13220910/182670384-912dcdf2-2296-4039-b977-fb5565a59505.png)

```r
library(tidyverse)
library(sf)
library(usmap)

nldas_metrics_all <- readr::read_csv('annual_metrics_glm3_pb0nldas.csv')

# Get data for mapping
site_info <- readr::read_csv('lake_metadata.csv') # You can download this from the same ScienceBase item
state_sf <- usmap::us_map(include = unique(site_info$state)) %>% 
  st_as_sf(coords = c('x', 'y')) 

# Map out lakes that have unexpectedly high winter values. Does it make sense?
high_winter_temp_val <- 18
high_winter_temp_sites <- nldas_metrics_all %>% 
  filter(max_surf_jan >= high_winter_temp_val | 
           max_surf_feb >= high_winter_temp_val | 
           max_surf_dec >= high_winter_temp_val) %>% 
  select(site_id) %>% 
  left_join(site_info) %>% 
  st_as_sf(coords = c('centroid_lon', 'centroid_lat'), crs = 4326) %>% 
  st_transform(crs = usmap::usmap_crs())

usmap::plot_usmap(include = unique(site_info$state), color = 'darkgrey', size=0.15) +
  geom_sf(data = high_winter_temp_sites, aes(color = lake_depth_m), alpha=0.7) +
  scico::scale_color_scico(palette = 'batlow', direction = -1) + 
  ggtitle(sprintf('Lakes with at least one max surface temperature in Dec, Jan, or Feb above %s degC', high_winter_temp_val),
          subtitle = sprintf('n = %s lakes', nrow(high_winter_temp_sites)))
```

**Reply by lindsayplatt** [reply]
Generally, the lakes that have really high summer temperatures are either in the southern part of the footprint and/or are shallow lakes.

![image](https://user-images.githubusercontent.com/13220910/182671498-c2f865c3-3344-4dc7-b3ff-fbdaedafe718.png)

```r
library(tidyverse)
library(sf)
library(usmap)

nldas_metrics_all <- readr::read_csv('annual_metrics_glm3_pb0nldas.csv')

# Get data for mapping
site_info <- readr::read_csv('lake_metadata.csv') # You can download this from the same ScienceBase item
state_sf <- usmap::us_map(include = unique(site_info$state)) %>% 
  st_as_sf(coords = c('x', 'y')) 

# Map out lakes that have unexpectedly high summer values. Does it make sense?
high_summer_temp_val <- 40
high_summer_temp_sites <- nldas_metrics_all %>% 
  filter(max_surf_jun > high_summer_temp_val | 
           max_surf_jul > high_summer_temp_val | 
           max_surf_aug > high_summer_temp_val) %>% 
  select(site_id) %>% 
  left_join(site_info) %>% 
  st_as_sf(coords = c('centroid_lon', 'centroid_lat'), crs = 4326) %>% 
  st_transform(crs = usmap::usmap_crs())

usmap::plot_usmap(include = unique(site_info$state), color = 'darkgrey', size=0.15) +
  geom_sf(data = high_summer_temp_sites, aes(color = lake_depth_m), alpha=0.7) +
  scico::scale_color_scico(palette = 'batlow', direction = -1) + 
  ggtitle(sprintf('Lakes with at least one max surface temperature in Jun, Jul, or Aug above %s degC', high_summer_temp_val),
          subtitle = sprintf('n = %s lakes', nrow(high_summer_temp_sites)))
```

**Comment by hcorson-dosch-usgs** [comment]
Okay I just re-ran the NLDAS models for the full footprint, following the [fix to delete failed simulation directories](https://github.com/USGS-R/lake-temperature-process-models/pull/77) and the update to the [nml_list.rds](https://github.com/USGS-R/lake-temperature-model-prep/pull/358). 

As Lindsay and I had anticipated, all of the NLDAS models rebuilt, but many of the extraction steps were skipped, as the output was identical for the majority of model runs.

Model output changed for 12 sites across 5 states (WI, MI, MN, SD, IL):
```r
> output_summary <- readr::read_csv('../../../Downloads/GLM_NLDAS_summary.csv',col_types=cols())
> changed_output <- output_summary %>% 
+   left_join(output_summary_old, by='site_id', suffix = c('_current','_old')) %>%
+   filter((export_fl_hash_current != export_fl_hash_old)) %>%
+   pull(site_id) %>%
+   dput()
c("nhdhr_{D8A9DD23-94AD-4C15-B242-FE519B5EDF8C}", "nhdhr_109946618", 
"nhdhr_116542403", "nhdhr_120018832", "nhdhr_144272629", "nhdhr_151949010", 
"nhdhr_152517749", "nhdhr_157360156", "nhdhr_48784080", "nhdhr_70333995", 
"nhdhr_75651917", "nhdhr_80997655")
```
As expected, the [two WI sites](https://github.com/USGS-R/lake-temperature-model-prep/issues/357) are among those with changed output.

And we have gained output for one additional site, in IN:
```r
> new_output <- output_summary %>% 
+   left_join(output_summary_old, by='site_id', suffix = c('_current','_old')) %>%
+   filter(is.na(export_fl_hash_old)) %>%
+   pull(site_id) %>%
+   dput()
"nhdhr_123165634"
```
So our success percentage is still 98.6% (NLDAS models for 12688/12864 lakes succeeded, models for 176 lakes failed)

**Reply by hcorson-dosch-usgs** [reply]
I currently have a job submitted to re-run the `p5_nldas` evaluation steps. The matched pred-obs and the evaluation plots will reflect these latest changes (if these sites are among those with sufficient observations) and the switch to [extrapolating to lake depth](https://github.com/USGS-R/lake-temperature-process-models/pull/75). Although the matched pred-obs have already been re-generated once following the extrapolation change, [by Lindsay](https://github.com/USGS-R/lake-temperature-process-models/issues/59#issuecomment-1194550488).

**Reply by hcorson-dosch-usgs** [reply]
The updated evaluation plots and matched pred-obs are [here](https://doimspp.sharepoint.com/:f:/r/sites/IIDDStaff/Shared%20Documents/Project%20-%20Modeling%20Lakes/GLM/lake-temperature-process-models/NLDAS_evaluation_plots/FullFootprint_20220826?csf=1&web=1&e=K16E6M). The # of evaluation sites didn't change, since the newly added site `nhdhr_123165634` didn't have >= 10 observation dates. 3 of the sites with updated GLM output are among the evaluation sites (`nhdhr_116542403`, `nhdhr_120018832`, `nhdhr_75651917`).

**Reply by hcorson-dosch-usgs** [reply]
The updated NLDAS output and matched pred-obs are [on SB](https://www.sciencebase.gov/catalog/item/6206d3c2d34ec05caca53071). Note that 5 of the NLDAS zips (for ND, MO, IA, AR, and OH) have not been updated, as no output changed for sites in those states.

**Comment by lindsayplatt** [comment]
Using the latest NLDAS-driven GLM build completed on August 29, 2022, the full footprint of thermal metrics has been rebuilt for the 12,688 successfully modeled lakes. These new NLDAS thermal metrics have been updated [on SB](https://www.sciencebase.gov/catalog/item/6206d3c2d34ec05caca53071).
